### ISSUE TITLE
restic find is not interrupted by a SIGINT (CTRL+C)

### ISSUE THREAD

**Issue by frabe1579** [restic/restic#5025]
Output of `restic version`
--------------------------
restic 0.17.0 compiled with go1.22.5 on windows/amd64

What backend/service did you use to store the repository?
---------------------------------------------------------
Local directory

Problem description / Steps to reproduce
----------------------------------------
The execution of `restic find` in a large repository is not interrupted by a SIGINT (CTRL+C).
When typing CTRL+C, restic outputs `signal interrupt received, cleaning up`, but it doesn't exit until completion of the full search.

The fact that the repository must be large is only useful for testing, in order to take some time to execute the command, and having time to press CTRL+C.

Try this command on a large repository:
```
restic find foo.bar
```
Output, pressing CTRL+C after some seconds:
```
repository xxxxxxxx opened (version 2, compression level auto)
[0:00] 100.00%  7 / 7 index files loaded
Found matching entries in snapshot yyyyyyyy from 2024-08-19 18:20:18
/d/zzzzzzzzzzzzzzz/foo.bar
/d/kkkkkkkkkkkkkk/foo.bar
signal interrupt received, cleaning up
/d/wwwwwwwwww/foo.bar
```
The command continues after pressing CTRL+C, and other find results are outputted after `signal interrupt received, cleaning up`

Expected behavior
-----------------
restic find is interrupted by a CTRL+C, exiting briefly.

Actual behavior
---------------
restic continues without exiting, as if it has not received SIGINT, also if outputted a message regarding it.

Do you have any idea what may have caused this?
-----------------------------------------------
No, other restic commands seem to work correctly.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes, every day. Using it for years with dozens of repositories for me and my work, also restoring Terabytes of data without any problem.

**Comment by frabe1579** [comment]
#4957 References `find`, but seems not resolved for it.

**Comment by MichaelEischer** [comment]
@frabe1579 The fix is not yet included in a released version. Either compile restic yourself or grab a beta version from https://beta.restic.net/?sort=time&order=desc .

**Comment by frabe1579** [comment]
Sorry, confused ISSUE with a PR merge.
Thanks.

**Comment by MichaelEischer** [comment]
Fix is included in restic 0.17.1
